metadata_backend/api/operators/file.py

```diff
@@ -193,18 +202,61 @@ async def read_project_files(self, project_id: str) -> List[Dict]:
             LOG.exception(reason)
             raise web.HTTPInternalServerError(reason=reason) from error
 
+    async def check_submission_files_ready(self, submission_id: str) -> None:
+        """Check all files in a submission are marked as ready.
+
+        Files marked as ready in a submission, means an metadata object has been
+        attached to the file.
+
+        :param submission_id: Submission ID to get associated files status
+        :raises: HTTPInternalServerError if db operation failed because of connection
+        or other db issue
+        """
+        aggregate_query = [
+            {"$match": {"submissionId": submission_id}},
+            {"$unwind": "$files"},
+            # check the status is not in failed or added
+            # failed can occour when an file ingestion/verification/mapping fails
+            {"$match": {"files.status": {"$in": ["added", "failed"]}}},
+            {
+                "$project": {
+                    "_id": 0,
+                    "accessionId": "$files.accessionId",
+                    "version": "$files.version",
+                    "status": "$files.status",
+                }
+            },
+        ]
+        try:
+            problematic_files = await self.db_service.do_aggregate("submission", aggregate_query)
+            if len(problematic_files) > 0:
+                reason = (
+                    f"There are a problematic files: {','.join([i['accessionId'] for i in problematic_files])} "
```

Comment:
Would the message say that the files are not in use, and should either be used or removed?

Or should we remove unlinked files in the publishing phase? ^^

Reply:
this is more of a check to log these files, however: i am thinking more also of a retry mechanism if they users keep the files.
So this function might change.

The update (remove or unlink the files) will be done via the submission API, but this needs to be an explicit action.